tfhe/src/integer/server_key/radix_parallel/scalar_sub.rs

```diff
@@ -112,4 +116,57 @@ impl ServerKey {
             self.full_propagate_parallelized(ct);
         }
     }
+
+    pub fn unsigned_overflowing_scalar_sub_parallelized<T>(
+        &self,
+        lhs: &RadixCiphertext,
+        scalar: T,
+    ) -> (RadixCiphertext, BooleanBlock)
+    where
+        T: UnsignedNumeric + DecomposableInto<u8>,
+    {
+        let mut tmp_lhs;
+
+        let lhs = if lhs.block_carries_are_empty() {
+            lhs
+        } else {
+            tmp_lhs = lhs.clone();
+            self.full_propagate_parallelized(&mut tmp_lhs);
+            &tmp_lhs
+        };
+
+        let mut scalar_decomposer =
+            BlockDecomposer::new(scalar, self.message_modulus().0.ilog2()).iter_as::<u8>();
+
+        let ct = lhs
+            .blocks
+            .iter()
+            .zip(scalar_decomposer.by_ref())
+            .map(|(lhs_block, rhs_scalar)| {
+                let mut result = lhs_block.clone();
+                let delta =
+                    (1_u64 << 63) / (self.key.message_modulus.0 * self.key.carry_modulus.0) as u64;
+
+                let encoded_scalar = Plaintext(rhs_scalar as u64 * delta);
+                lwe_ciphertext_plaintext_sub_assign(&mut result.ct, encoded_scalar);
+
+                let encoded_msg_mod = Plaintext(self.key.message_modulus.0 as u64 * delta);
+                lwe_ciphertext_plaintext_add_assign(&mut result.ct, encoded_msg_mod);
+
+                result.degree += Degree::new(self.key.message_modulus.0);
+                result
```

Comment:
is that a scalar with correcting term ? I'm surprised we don't have that in shortint 🤔

Reply:
Nope it does not exists

Reply:
do you think it could be interesting to add ?

Reply:
I moved that part into its  own unchecked_scalar_sub_with_correcting_term